packages/eslint-plugin/src/rules/no-useless-template-literals.ts

```diff
@@ -68,6 +72,27 @@ export default createRule<[], MessageId>({
           context.report({
             node: node.expressions[0],
             messageId: 'noUselessTemplateLiteral',
+            suggest: [
```

Comment:
[Question] Is there a reason to have this as a `suggest`ion, rather than a `fix`? It shouldn't change code or types (roughly the only dividing line between fixes and suggestions) that I can see?

Reply:
no special reason other than that this is what we've done [here](https://github.com/typescript-eslint/typescript-eslint/pull/7691), so I didn't really put much thought into that.

can probably be a fix 😄 

how do you test multiple fixes in the same line though?

```TS
`${'a'}${true}`;
```

Reply:
The fixer function can be a generator / return an array of fixes:

https://github.com/typescript-eslint/typescript-eslint/blob/99a026fe41b4b169914352fe66e7d5d687b300bf/packages/utils/src/ts-eslint/Rule.ts#L114-L116

A couple of examples of that:
* `no-unnecessary-boolean-literal-compare` is a `fix: function* (fixer) {`
* `array-type` has a couple fixers that return arrays

Is that what you're looking for?

Reply:
Yeah, I know it can return an array, but each of those nodes is being reported separately, so I thought that eslint would fix everything before returning the result in the test.

[The docs](https://eslint.org/docs/latest/extend/custom-rules#applying-fixes) state that:

> After applying fixes, ESLint will run all the enabled rules again on the fixed code, potentially applying more fixes. This process will repeat up to 10 times, or until no more fixable problems are found.

So I need to report on all of the nodes together? (if so, it'll put the squiggly line on the whole template literal which is probably not what we want)

Reply:
Ah gotcha, sorry, I think I see what you mean now. Since each report is about a single violation I think each report should only suggest fixing that violation. They won't overlap ranges so ESLint should be able to apply all the fixes in one pass.

I don't remember whether ESLint will hold off applying fixes that border each other (e.g. how <code>\`${'a'}${'b'}\`</code> might report)... but if that is the case, ESLint would still be able to apply alternating fixes. So it should be no more than two fix passes.

I.e. what I'm suggesting is roughly this psuedocode-ish:

```ts
context.report({
  loc: { end, start },
  fix: (fixer) => replaceRange(end, start),
});
```

Reply:
if I run it locally, it auto-fixes both of them on save (not sure if it's in a single pass though), so theoretically it should be fine and (IMO) should behave the same in tests, no?

I mean... do we need to change the otherwise-working code only because the tests are not working?
maybe we should fix the rule tester to behave the same as in the real world? (or am I missing something here?)

Not sure what you mean in the pseudo-code BTW

Reply:
OK, after an investigation, seems like this is the issue:
https://github.com/typescript-eslint/typescript-eslint/blob/2e7ce16344253e530deddce2ebf044403342dc85/packages/rule-tester/src/utils/SourceCodeFixer.ts#L65-L69

The fixer in the tests _should_ fix everything, but we check for `lastPos >= start`, which in my test cases was true after the first fix.
I considered changing it to `lastPos > start`, but wasn't sure if it's OK or not (only 2 tests were failing, which I think had the same issue)

Anyway, adding a space between the 2 errors seems to fix the issue (pun intended)
(e.g. `${'a'}${'b'}` => `${'a'} ${'b'}`)

Reply:
Oh, is this a bug in RuleTester then? That might be a separate issue.

Reply:
I think it is, but it probably requires more investigation, will raise an issue after this PR.
It's ready for review btw, I can't re-request review for some reason

Reply:
Filed a new issue:
https://github.com/typescript-eslint/typescript-eslint/issues/8251